Kernel_23/include/CGAL/Bbox.h

```diff
@@ -0,0 +1,159 @@
+    {
+        CGAL_assertion(min_values.size() == 0 || min_values.size() == bbox.min_values.size());
+        int dim = bbox.min_values.size();
+        for(int i=0; i<dim; ++i)
+        {
+            if(min_values[i] > bbox.min_values[i]) min_values[i] = bbox.min_values[i];
+            if(max_values[i] < bbox.max_values[i]) max_values[i] = bbox.max_values[i];
+        }
+        return *this;
+    }
+
+    inline int dimension() const
+    {
+        return static_cast<const Derived*>(this)->dimension();
+    }
+
+    inline T min BOOST_PREVENT_MACRO_SUBSTITUTION (int i) const
+    {
+        return min_values[i];
+    }
+
+    inline T max BOOST_PREVENT_MACRO_SUBSTITUTION (int i) const
+    {
+        return max_values[i];
+    }
+
+    inline T& min BOOST_PREVENT_MACRO_SUBSTITUTION (int i)
+    {
+        return min_values[i];
+    }
+
+    inline T& max BOOST_PREVENT_MACRO_SUBSTITUTION (int i)
+    {
+        return max_values[i];
+    }
+
+    inline T measure() const {
+        T result = max_values[0] - min_values[0];
+        if (result <= 0) return 0;
+        for(int i=1; i<dimension(); ++i)
+            result *= max_values[i] - min_values[i];
+        return result;
+    }
+
+    inline T intersection_measure(const Bbox& bbox) const {
+        CGAL_assertion(dimension() == bbox.dimension());
+        T result = 1;
+        for(int i=0; i<dimension(); ++i) {
+            result *= std::min(max(i), bbox.max(i)) -
+                      std::max(min(i), bbox.min(i));
+            if (result <= 0) return 0;
+        }
+        return result;
+    }
+
+    bool operator==(const Bbox& bbox) const {
+        for(int i=0; i<dimension(); ++i)
+            if (min_values[i] != bbox.min_values[i] || max_values[i] != bbox.max_values[i])
+                return false;
+        return true;
+    }
+
+    bool operator!=(const Bbox& bbox) const { return !operator==(bbox); }
+
+protected:
+    void init(int d, T range = -std::numeric_limits<T>::infinity()) {
+        for(int i=0; i<d; ++i)
+        {
+            min_values[i] = -range;
+            max_values[i] =  range;
+        }
+    }
+};
+
+}
+
+// A D-dimensional axis aligned box
+template<unsigned int N, typename T>
+class Bbox : public Impl::Bbox<std::array<T, N>, Bbox<N,T>>
+{
+    enum { D = N };
+public:
+    inline constexpr int dimension() const { return D; }
+    Bbox(int d = 0          ) { assert(d==N || d==0); this->init(d       ); }
+    Bbox(int d, double range) { assert(d==N || d==0); this->init(d, range); }
+};
+
+// A D-dimensional axis aligned box
+template<typename T>
+class Bbox<0,T> : public Impl::Bbox<std::vector<T>, Bbox<0,T>>
+{
+public:
+    inline int dimension() const { return this->min_values.size(); }
+    Bbox(int d = 0          ) { init_values(d); this->init(d       ); }
+    Bbox(int d, double range) { init_values(d); this->init(d, range); }
+
+protected:
+    void init_values(int d) {
+        this->min_values.resize(d);
+        this->max_values.resize(d);
```

Comment:
It seems a bit wasteful to use 2 containers of size d instead of one of size 2*d, but that's an implementation detail so it doesn't matter now.